### ISSUE TITLE
[DataGrid] error on edit `Filled Quantity` column

### ISSUE THREAD

**Issue by siriwatknp** [mui-org/material-ui-x#2556]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

error after trying to edit `Filled Quantity` column (using data-generator package). https://deploy-preview-28106--material-ui.netlify.app/branding/x-advanced/
<img width="1440" alt="Screen Shot 2564-09-06 at 14 30 52" src="https://user-images.githubusercontent.com/18292247/132177552-c293d5fa-5ea4-4ed2-9f24-6060392463af.png">

using

```json
 "@material-ui/x-grid": "^4.0.0-alpha.37",
 "@material-ui/x-grid-data-generator": "^4.0.0-alpha.37",
```

### Expected behavior 🤔

should not throw error

### Steps to reproduce 🕹

Steps:

1. goto https://deploy-preview-28106--material-ui.netlify.app/branding/x-advanced/
2. on the hero section, scroll to the right and find `Filled quantity` column
3. try to edit some row
4. error occured

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by m4theushw** [comment]
I noticed that in https://github.com/mui-org/material-ui-x/pull/2441#issuecomment-912647413 as well. We need to update `@mui/x-grid-data-generator` to support v5.

**Comment by m4theushw** [comment]
Fixed by #2571
